src/Specular/FRP/Replaceable.purs

```diff
@@ -25,3 +32,27 @@ weakDynamic :: forall m a. MonadReplace m => MonadFRP m => WeakDynamic (m a) ->
 weakDynamic dyn = do
   Slot {replace} <- newSlot
   subscribeWeakDyn (\x -> replace x) dyn
+
+whenJustD :: forall m a. MonadReplace m => MonadFRP m => Dynamic (Maybe a) -> (Dynamic a -> m Unit) -> m Unit
+whenJustD dyn widget = do
+  shouldDisplay <- uniqDynBy eq (isJust <$> dyn)
+  whenD shouldDisplay do
+    dynVal <- readDynamic dyn
+    case dynVal of
+      Just dynVal' -> do
+        dyn' <- newDynamic dynVal'
+        subscribeEvent_ dyn'.set $ filterJustEvent $ changed dyn
+        widget dyn'.dynamic
+      Nothing -> pure unit
+
+-- | Execute a monadic action in a Replaceable monad only if the given Dynamic
```

Comment:
It's not clear from this comment that the action will be also undone when the dynamic turns false, and redone when it turns true again.

Although I can't think of a better doc right now.